### PR TITLE
Jetpack Forms: Add tracking of Google Sheets exports of responses

### PR DESCRIPTION
**Pull request by donnchawp**
Dupe of #29177

When responses are exported from WP-Admin->Feedback->Form Responses they can be exported to Google Sheets. This patch adds tracking of those actions.

It requires the record_tracks_event function from https://github.com/Automattic/jetpack/pull/29102.

Discussion: pcsBup-S8-p2
Testing instructions:

Wait for https://github.com/Automattic/jetpack/pull/29102 to be merged or grab record_tracks_event from it.
Create a form block in a post with /form.
Fill out the form and submit.
Go to WP-Admin->Feedback->Form Responses and export the response(s) to Google Sheets. You'll have to connect it first. If using Firefox containers you might have to do this bit in Chrome.
If an Automattician, look for the forms_export_responses tracking event. The format prop should be "gsheets"
Does this pull request change what data or activity we track or use?

Yes, it tracks exports by Google Sheets from the Feedback Forms page.

Have you written new tests for your changes, if applicable?
Have you checked the E2E test CI results, and verified that your changes do not break them?
Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

### CODE REVIEW

**Review by CGastrell — DISMISSED** [review]
Still looking good :shipit:

File: projects/packages/forms/src/contact-form/class-admin.php [hunk 1]
```diff
@@ -223,6 +223,9 @@ public function export_to_gdrive() {
 
 		$sheet = Google_Drive::create_sheet( $user_id, $spreadsheet_title, $sheet_data );
 
+		$tracking = new Automattic\Jetpack\Tracking();
```
Comment (CGastrell) [inline projects/packages/forms/src/contact-form/class-admin.php]: As this is a namespaced class/file, please use root path for the class name, something similar was done on https://github.com/Automattic/jetpack/pull/29102:

```php
$tracking = new \Automattic\Jetpack\Tracking();
```

File: projects/plugins/jetpack/modules/contact-form/admin.php [hunk 2]
```diff
@@ -1319,6 +1319,9 @@ public function export_to_gdrive() {
 		require_once JETPACK__PLUGIN_DIR . '_inc/lib/class-jetpack-google-drive-helper.php';
 		$sheet = Jetpack_Google_Drive_Helper::create_sheet( $user_id, $spreadsheet_title, $sheet_data );
 
+		$tracking = new Automattic\Jetpack\Tracking();
```
Comment (CGastrell) [inline projects/plugins/jetpack/modules/contact-form/admin.php]: While this is NOT a namespaced file, we could also use the syntax mentioned above, just for the sake of consistency.

**Review by CGastrell — CHANGES_REQUESTED** [review]
The call on for `Automattic\Jetpack\Tracking` would fail on the package as the file is namespaced, hence, the path/scope is wrongly interpreted.

File: projects/packages/forms/src/contact-form/class-admin.php (hunk 1 above)
Comment (donnchawp) [reply projects/packages/forms/src/contact-form/class-admin.php]: Thanks, just fixed it!

**Comment by jeherve** [comment]
Changes have been addressed, I'm updating the labels and requesting a new review to keep this PR moving.

**Review by jeherve — DISMISSED** [review]
This tests well for me. 👍

**Review by CGastrell — DISMISSED** [review]
LGTM :shipit:

**Comment by donnchawp** [comment]
In my last commit I changed it back to using record_tracks_event so it works everywhere, Jetpack and WordPress.com. This means that #29102 must be merged before this PR is merged.

**Comment by donnchawp** [comment]
#29102 was merged this morning so we can test this now.

**Review by jeherve — DISMISSED** [review]
I don't think your branch is up to date; it doesn't include the last commit that would help us here:
https://github.com/Automattic/jetpack/commits/add/tracking_google_sheets_exports

As a result, you would get a Fatal error if you were to test this.

**Comment by donnchawp** [comment]
I did a `git merge` but I noticed something I thought I had changed in the other PR. The record_tracks_event function must be public to be used by this PR. I'll fix that now.

**Review by CGastrell — DISMISSED** [review]
Still looking good! :shipit:

**Review by jeherve — APPROVED** [review]

**Pull request merged**